pester.go

```diff
@@ -165,6 +171,14 @@ func LinearJitterBackoff(i int) time.Duration {
 	return jitter(i)
 }
 
+// DefaultRetryStrategy returns true in the following case:
+//  - status code is lower than http.StatusInternalServerError and
+//  - status is not http.StatusTooManyRequests or status is http.StatusTooManyRequests with retryOnHTTP429 flag set false
+func DefaultRetryStrategy(status int, retryOnHTTP429 bool) bool {
+	return status < http.StatusInternalServerError &&
+		(status != http.StatusTooManyRequests || (status == http.StatusTooManyRequests && !retryOnHTTP429))
+}
```

Comment:
What do you think about making `RetryStrategy` simply `func(status int) bool` and then changing this to

```suggestion
// MakeDefaultRetryStrategy creates a function which returns true in the following case:
//  - status code is lower than http.StatusInternalServerError and
//  - status is not http.StatusTooManyRequests or status is http.StatusTooManyRequests with retryOnHTTP429 flag set false
func MakeDefaultRetryStrategy(retryOnHTTP429 bool) RetryStrategy {
	return func(status int) bool {
		status < http.StatusInternalServerError &&
			(status != http.StatusTooManyRequests || (status == http.StatusTooManyRequests && !retryOnHTTP429))
	}
}
```

Note - not sure whether I got the syntax right here.

Reply:
I found this function a bit har to read easily, so what do you thing about switching to simple if cases?

```
// DefaultRetryStrategy returns true in the following case:
//  - status code is lower than http.StatusInternalServerError and status is not http.StatusTooManyRequests
//  - status code is lower than http.StatusInternalServerError and status is http.StatusTooManyRequests 
// 	with retryOnHTTP429 flag set to false
func DefaultRetryStrategy(status int, retryOnHTTP429 bool) bool {
	if status < http.StatusInternalServerError && status != http.StatusTooManyRequests {
		return true
	}
	if status < http.StatusInternalServerError && status == http.StatusTooManyRequests && !retryOnHTTP429 {
		return true
	}
	return false
}
```

Reply:
> what do you thing about switching to simple if cases?

Sounds good to me.
But IMO it should still be:
```
func MakeDefaultRetryStrategy(retryOnHTTP429 bool) RetryStrategy {
	return func(status int) bool {
		// default implementation
	}
}
```

For backwards compatibility reasons and to make sure that public API is _somewhat sane_.

Reply:
// DefaultRetryStrategy returns true in the following case:
//  - status code is lower than http.StatusInternalServerError and
//  - status is not http.StatusTooManyRequests or status is http.StatusTooManyRequests with retryOnHTTP429 flag set false
func DefaultRetryStrategy(status int, retryOnHTTP429 bool) bool {
	return status < http.StatusInternalServerError &&
		(status != http.StatusTooManyRequests || (status == http.StatusTooManyRequests && !retryOnHTTP429))
}
```

Reply:
I played with Github features so the review currently looks horrible, my updated comment which you replied to is now below your response. Also this "what do you thing" typo is triggering me :D

Reply:
I will try that.

Reply:
I tried all possible ways to make this similar to the what Vedran suggested, but failed. It's impossible to make 2 parameter function 1 parameter, and still keep it simple and easy to use.

I still think that one function with 2 parameters and simple bool return is the simplest solution to this problem.